### ISSUE TITLE
Feature request: sort output by line

### ISSUE THREAD

**Issue by Jmuccigr** [markdownlint/markdownlint#157]
It would be great to have the output sorted by line, instead of by rule. It would be easy enough to do this by piping through `sort` if the line numbers all had the same length (so, `001` instead of `1`).

**Comment by nbehrnd** [comment]
@Jmuccigr In case of Linux Debian and its default shell `bash`, it is possible to relay (or better: pipe) the result by mdl to `sort` of the GNU coreutils.  This runs in a pattern of

```shell
mdl ./README.md | sort -t ':' -k 2 -n
```

where `-t ':'` indicates the characters I would consider as «column separator» to be the colon `:`.  Further on, the key/criterion to perform the sort is the now defined second column (containing the line number), `-k 2`.  To get line 2 reported prior to line 11, one opts-in for a *natural* sort instead of the default one which advances character by character.

With version 0.13.0 tested (as Debian 13/trixie's `markdownlint`), an analysis basically runs as

```shell
$ mdl ./README.md | sort -t ':' -k 2 -n

Further documentation is available for these failures:
 - MD007: https://github.com/markdownlint/markdownlint/blob/main/docs/RULES.md#md007---unordered-list-indentation
 - MD012: https://github.com/markdownlint/markdownlint/blob/main/docs/RULES.md#md012---multiple-consecutive-blank-lines
 - MD013: https://github.com/markdownlint/markdownlint/blob/main/docs/RULES.md#md013---line-length
 - MD014: https://github.com/markdownlint/markdownlint/blob/main/docs/RULES.md#md014---dollar-signs-used-before-commands-without-showing-output
 - MD022: https://github.com/markdownlint/markdownlint/blob/main/docs/RULES.md#md022---headers-should-be-surrounded-by-blank-lines
 - MD032: https://github.com/markdownlint/markdownlint/blob/main/docs/RULES.md#md032---lists-should-be-surrounded-by-blank-lines
 - MD034: https://github.com/markdownlint/markdownlint/blob/main/docs/RULES.md#md034---bare-url-used
./README.md:3: MD013 Line length
./README.md:6: MD032 Lists should be surrounded by blank lines
./README.md:9: MD013 Line length
./README.md:11: MD012 Multiple consecutive blank lines
./README.md:14: MD013 Line length
./README.md:30: MD013 Line length
./README.md:32: MD013 Line length
./README.md:34: MD012 Multiple consecutive blank lines
./README.md:41: MD013 Line length
./README.md:43: MD007 Unordered list indentation
./README.md:43: MD013 Line length
./README.md:50: MD014 Dollar signs used before commands without showing output
./README.md:56: MD013 Line length
./README.md:60: MD013 Line length
./README.md:67: MD012 Multiple consecutive blank lines
```
